crawler/src/main.rs

```diff
@@ -0,0 +1,72 @@
+use colored::*;
+use serde::{Deserialize, Serialize};
+
+type Listing = ProxyData<Posts>;
+
+#[derive(Debug, Deserialize, Serialize)]
+struct ProxyData<T> {
```

Comment:
Why is this generic instead of just containing `pub data: Posts`?

Reply:
I use it at multiple places, in the Listing but also for the children.

Reply:
Ah, yes, I see it now.